File: Java Based Archives/StudentRegister/app/src/main/java/com/android/studentregister/entity/Student.java
package com.android.studentregister.entity;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "student_table")
public class Student extends BaseObservable {
    @PrimaryKey(autoGenerate = true)
    private int studentId;
    private String name;
    private String email;
    private String country;
    private String registeredTime;

    // Default constructor
    @Ignore
    public Student() {
    }

    // Parameterized constructor to initialize fields
    public Student(int studentId, String name, String email, String country, String registeredTime) {
        this.studentId = studentId;
        this.name = name;
        this.email = email;
        this.country = country;
        this.registeredTime = registeredTime;
    }

    // Getters and Setters
    @Bindable
    public int getStudentId() {
        return studentId;
    }

    public void setStudentId(int studentId) {
        this.studentId = studentId;
    }

    @Bindable
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;

    }

    @Bindable
    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;

    }

    @Bindable
    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;

    }

    @Bindable
    public String getRegisteredTime() {
        return registeredTime;
    }

    public void setRegisteredTime(String registeredTime) {
        this.registeredTime = registeredTime;

    }
}

File: Views/RoomMigrationDemo1/app/src/main/java/com/anushka/roommigrationdemo1/Student.kt
package com.anushka.roommigrationdemo1

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "student_info")
data class Student(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "student_id")
    val id: Long,

    @ColumnInfo(name = "student_name", defaultValue = "No Name")
    val name: String,

    @ColumnInfo(name = "student_email", defaultValue = "No Email")
    val email: String?,

    @ColumnInfo(name = "subject_name", defaultValue = "No subject")
    val course: String
)

